dev-util/pkgconf/patches/pkgconf-1.4.2.patchset

```diff
@@ -0,0 +1,92 @@
+From a228737146c07f0b4d6f53916d64a8a65e405fda Mon Sep 17 00:00:00 2001
+From: Leorize <alaviss@users.noreply.github.com>
+Date: Tue, 3 Apr 2018 12:46:35 +0700
+Subject: libpkgconf: add support for Haiku
+
+client: use BELIBRARIES
+pkg: bootstrap package search paths with Haiku's find_paths
+
+diff --git a/libpkgconf/client.c b/libpkgconf/client.c
+index 811e043..a832e83 100644
+--- a/libpkgconf/client.c
++++ b/libpkgconf/client.c
+@@ -78,7 +78,11 @@ pkgconf_client_init(pkgconf_client_t *client, pkgconf_error_handler_func_t error
+ 	pkgconf_path_build_from_environ("PKG_CONFIG_SYSTEM_INCLUDE_PATH", SYSTEM_INCLUDEDIR, &client->filter_includedirs, false);
+ 
+ 	/* GCC uses these environment variables to define system include paths, so we should check them. */
++#ifdef __HAIKU__
++	pkgconf_path_build_from_environ("BELIBRARIES", NULL, &client->filter_libdirs, false);
++#else
```

Comment:
```
#ifdef __HAIKU__
	pkgconf_path_build_from_environ("BELIBRARIES", NULL, &client->filter_libdirs, false);
#else
	pkgconf_path_build_from_environ("LIBRARY_PATH", NULL, &client->filter_libdirs, false);
#endif
```
I don't understand why you skip the `LIBRARY_PATH` when `__HAIKU__` is defined. `LIBRARY_PATH` **_is_** used in Haiku. (Actually, if I'm not mistaken, Haiku's `LIBRARY_PATH` is the equivalent of Linux's `LD_LIBRARY_PATH`.)
Well, I might be missing something, so maybe there is no error here.

Reply:
That's the problem, Haiku uses LIBRARY_PATH in the runtime_loader for a purpose similar to LD_LIBRARY_PATH in Linux. So if you change LIBRARY_PATH, some apps will stop running because they don't find their runtime libraries.

In Linux, LIBRARY_PATH is used to tell the dev tools where to look for libraries. This is indeed somewhat similar to our BELIBRARIES.

Reply:
Thanks a lot, @pulkomandy, for this very important info.